packages/bot-web-ui/src/pages/server-side-bot/add-bot/add-bot.tsx

```diff
@@ -0,0 +1,241 @@
+                            min = +form_data?.stake;
+                            if (isNaN(min)) {
+                                min = +(initial_value?.stake ?? 0);
+                            }
+                            min_error = getErrorMessage('MIN', min);
+                        }
+                        if (should_validate && field.name === 'last_digit_prediction') {
+                            min = 0;
+                            max = 9;
+                            max_error = getErrorMessage('MAX', max, 'LAST_DIGIT_PREDICTION');
+                            integer_error_message = 'Enter a value from 0 to 9.';
+                        }
+                        if (should_validate) {
+                            field.validation.forEach(validation => {
+                                if (typeof validation === 'string') {
+                                    switch (validation) {
+                                        case 'required':
+                                            schema = schema.required(localize('Field cannot be empty'));
+                                            break;
+                                        case 'min':
+                                            schema = schema.min(min, min_error);
+                                            break;
+                                        case 'max':
+                                            schema = schema.max(max, max_error);
+                                            break;
+                                        case 'ceil':
+                                            schema = schema.round('ceil');
+                                            break;
+                                        case 'floor':
+                                            schema = schema.round('floor');
+                                            break;
+                                        case 'integer':
+                                            schema = schema.integer(integer_error_message);
+                                            break;
+                                        default:
+                                            break;
+                                    }
+                                } else if (typeof validation === 'object') {
+                                    if (validation?.type) {
+                                        schema = schema[validation.type](
+                                            validation.value,
+                                            localize(validation.getMessage(validation.value))
+                                        );
+                                    }
+                                }
+                            });
+                        }
+                        sub_schema[field.name] = schema;
+                    }
+                }
+            });
+        });
+        if (is_mounted.current) {
+            setDynamicSchema(Yup.object().shape(sub_schema));
```

Comment:
```suggestion
            setDynamicSchema(Yup.object().shape(sub_schema));
```
this dynamic setting of schema should be done in the useEffect - because it depends on the render cycle

Reply:
@yashim-deriv We are calling the `getErrors` function from the `validate` callback in the `useFormik` hook. This setup ensures that the validate function is triggered whenever there are changes in the field values that we need to validate and dynamically update the schema based on these specific cases.